### PR TITLE
Update the raft-otel project README

### PR DESCRIPTION
**Pull request by lauranooooo**
Some updates to this project to make it more approachable.
Primarily, changes to refer to a ready-made scaffolded app suitable for running in docker-compose.

### CODE REVIEW

File: projects/raft-otel/README.md [hunk 1]
```diff
@@ -97,50 +97,41 @@ Read these blog posts carefully.
...
+[CompareAndSwap](https://en.wikipedia.org/wiki/Compare-and-swap) (also called CompareAndSet, or CAS)
 is a very useful pattern for concurrent systems that lets you update a key to a given value only if that key already has a specific value.
 This is useful for implementing sequences of operations where we read a value, perform some computation that modifies that value, and then write that value back -
 but without potentially overwriting any changes to that value that other processes might have performed.
 
-Next, write a client that uses your RAFT cluster to perform Sets, Gets, and CompareAndSets.
-Bendersky's code also doesn't do anything if you send a `Submit` to any server other than the leader. It may be useful to have your program return the
-address (host:port) of the leader instead, as part of your gRPC reply, along with an indication that the operation was not attempted. Your client can then
-retry the operation against the leader.
+Update the demo client to use the new operation.
```
Comment (illicitonion) [inline projects/raft-otel/README.md:115]:
```suggestion
Add support for CompareAndSwap to the server implementation, and update the demo client to use the new operation after the existing operations it performs.
```

File: projects/raft-otel/README.md [hunk 2]
```diff
@@ -97,50 +97,41 @@ Read these blog posts carefully.
...
+
+##### Operations when an instance is not the leader - redirecting or proxying clients
+
+The initial code  doesn't do anything if you send a get/set to any server other than the leader. It may be useful to have your program return the
+address (host:port) of the leader instead, as part of your gRPC reply. Your client can then
+retry the operation against the leader. Alternatively, you could modify the program to proxy the request directly to the leader.
 
 Please timebox this part of the project to no more than two days, in order to leave time for the other sections.
 At the start of the third day, if your implementation is not complete, begin modifying Bendersky's. You can come back and complete your
 own implementation if you have time at the end of the sprint.
```
Comment (illicitonion) [inline projects/raft-otel/README.md:125]: Is this block meant to apply to this feature? It feels like at least the "Stop implementing your own raft" is misplaced here, maybe the entire para is?

File: projects/raft-otel/README.md [hunk 3]
```diff
@@ -97,50 +97,41 @@ Read these blog posts carefully.
...
 
+##### Operations when an instance is not the leader - allow stale reads
+
+The reason we do not serve read operations (here, `get`) from hosts that are not the RAFT leader is that they might have stale data.
+You could update the K/V Get operation to include parameters to let the client do stale reads from non-leaders.
+Ideally, you would allow the client to specify a duration of allowed staleness, and check whether the instance has successfully
+processed an incoming AppendEntries message from the leader within that duration.
```
Comment (illicitonion) [inline projects/raft-otel/README.md:132]:
```suggestion
processed an incoming AppendEntries message from the leader within that duration. The response from the `get` should probably include an indication of whether the read is stale.
```

File: projects/raft-otel/README.md [hunk 4]
```diff
@@ -162,8 +155,9 @@ Create a [Board](https://docs.honeycomb.io/working-with-your-data/boards/) in ho
 
 ### Part 3: Debugging latency and failures using distributed tracing
 
-Bendersky's implementation of RAFT allows you to simulate unreliable RPCs. Enable this (if you wrote your own implementation then you will
-need to add this kind of chaos capability as a feature first - for example, adding an environment variable which, if set, will drop some percent of requests completely, and add random latency to others).
+Add an environment variable which, if set, will drop some percentage of the internal RAFT requests completely, and add latency to others.
```
Comment (illicitonion) [inline projects/raft-otel/README.md:158]:
```suggestion
Add an environment variable which, if set, will drop some percentage of the internal RAFT requests (i.e. requests between RAFT servers) completely, and add latency to others.
```

**Review by illicitonion — APPROVED** [review]
I haven't looked at the scaffold code yet, but the exercise edits LGTM!

File: projects/raft-otel/README.md (hunk 2 above)
Comment (lauranooooo) [reply projects/raft-otel/README.md]: mvoed this to top.

**Review by illicitonion — APPROVED** [review]
Ship it!

**Pull request merged**